libraries/AP_FETtecOneWire/AP_FETtecOneWire.h

```diff
@@ -0,0 +1,270 @@
+    @return the current used ID
+*/
+    uint8_t init_escs();
+
+/**
+    sets the telemetry mode to full mode, where one ESC answers with all telem values including CRC Error count and a CRC
+    @param active if full telemetry should be used
+    @return returns the response code
+*/
+    uint8_t set_full_telemetry(uint8_t active);
+
+#if HAL_WITH_ESC_TELEM
+/**
+    checks if the requested telemetry is available.
+    @param t telemetry datastructure where the read telemetry will be stored in.
+    @param centi_erpm 16bit centi-eRPM value returned from the ESC
+    @param tx_err_count Ardupilot->ESC communication CRC error counter
+    @param tlm_from_id receives the ID from the ESC that has respond with its telemetry
+    @return 1 if CRC is correct, 2 on CRC mismatch, 0 on waiting for answer
+*/
+receive_response decode_single_esc_telemetry(TelemetryData& t, int16_t& centi_erpm, uint16_t& tx_err_count, uint8_t &tlm_from_id);
+#endif
+
+/**
+    scans for ESCs if not already done.
+    initializes the ESCs if not already done.
+    sends fast throttle signals if init is complete.
+    @param motor_values a 16bit array containing the throttle signals that should be sent to the motors. 0-2000 where 1001-2000 is positive rotation and 999-0 reversed rotation
+    @param motor_count the count of motors that should get values send
+    @param tlm_request the ESC to request telemetry from (0 for no telemetry, 1 for ESC0, 2 for ESC1, 3 for ESC2, ...)
+*/
+    void escs_set_values(const uint16_t *motor_values, const uint8_t motor_count, const uint8_t tlm_request);
+
+    typedef struct FETtecOneWireESC
+    {
+      uint8_t in_boot_loader;
+#if HAL_AP_FETTEC_ONEWIRE_GET_STATIC_INFO
+      uint8_t firmware_version;
+      uint8_t firmware_sub_version;
+      uint8_t esc_type;
+      uint8_t serial_number[12];
+#endif
+    } FETtecOneWireESC_t;
+
+    uint16_t _error_count[MOTOR_COUNT_MAX]; //saves the error counter from the ESCs
+    uint16_t _error_count_since_overflow[MOTOR_COUNT_MAX]; //saves the error counter from the ESCs to pass the overflow
+    uint16_t _send_msg_count; //counts the messages that are send by fc
+    uint16_t _update_loop_decimator;
+    uint16_t _mask;
+    uint8_t _nr_escs_in_bitmask;
+    uint16_t _update_rate_hz = 400;
+    float _crc_error_rate_factor;
+
+    uint8_t _active_esc_ids[MOTOR_COUNT_MAX] = {0};
+    FETtecOneWireESC_t _found_escs[MOTOR_COUNT_MAX];
+    uint8_t _found_escs_count;
+    uint8_t _scan_active;
+    uint8_t _setup_active;
+
+    uint8_t _set_full_telemetry_active = 1; //Helper to set alternative TLM for every ESC
+    uint8_t _set_full_telemetry_retry_count = 0; 
+    uint8_t _ignore_own_bytes;
+    int8_t _min_id = MOTOR_COUNT_MAX;
+    int8_t _max_id;
+    uint8_t _id_count;
+    uint8_t _fast_throttle_byte_count;
+    bool _pull_success;
+    bool _pull_busy;
+    uint8_t _tlm_request;
+    uint8_t _last_crc;
+    uint32_t _lastESCScan;
+    enum msg_type
+    {
+      OW_OK,

```

Comment:
If the protocol requires these to have specific integer values then you should put them in explicitly here